rcgen/src/lib.rs

```diff
@@ -865,6 +908,22 @@ impl CertificateParams {
 							SanType::IpAddress(IpAddr::V6(addr)) => {
 								writer.write_bytes(&addr.octets())
 							},
+							SanType::OtherName((oid, value)) => {
+								// otherName SEQUENCE { OID, [0] explicit any defined by oid }
+								// https://datatracker.ietf.org/doc/html/rfc5280#page-38
+								let oid = ObjectIdentifier::from_slice(&oid);
+								writer.write_sequence(|writer| {
+									writer.next().write_oid(&oid);
+									writer.next().write_tagged(
+										Tag::context(0),
+										|writer| match value {
+											OtherNameValue::Utf8String(s) => {
+												writer.write_utf8_string(s)
+											},
+										},
+									);
+								});
+							},
```

Comment:
A couple suggestions here:

* let's inline the `oid` binding - it doesn't get referenced outside of the one call to `write_oid`.

* I think to make it easier to extend the `OtherNameValue` enum it might make sense to pull out the code for serializing the value into a `write_der(&self, writer: DERWriter)` fn on `OtherNameValue`. We've been doing that with other types (e.g. `CrlDistributionPoint::write_der`, `PublicKeyData::serialize_public_key_der`) and I think it makes for easier to follow code.

Something like:

```
impl OtherNameValue {
	fn write_der(&self, writer: DERWriter) {
		writer.write_tagged(Tag::context(0), |writer| match self {
			OtherNameValue::Utf8String(s) => writer.write_utf8_string(s),
		});
	}
}
``` 

which would let us write this section as:

```suggestion
								// otherName SEQUENCE { OID, [0] explicit any defined by oid }
								// https://datatracker.ietf.org/doc/html/rfc5280#page-38
								writer.write_sequence(|writer| {
									writer.next().write_oid(&ObjectIdentifier::from_slice(&oid));
									value.write_der(writer.next());
								});
```

It's a bit unfortunate that we define `TAG_OTHER_NAME` as part of `SanType::tag()` but can't use it in the serialization of the other name value (instead it's hardcoded as `0`), but I think addressing that will require a larger refactor that pulls out the san serialization into a fn on `SanType` and that feels like too much to foist on you here.

Reply:
Thanks that's cleaner this way